packages/block-library/src/navigation/edit.js

```diff
@@ -41,15 +46,27 @@ function Navigation( {
 	hasItemJustificationControls = true,
 	hasListViewModal = true,
 } ) {
+	const navIcons = {
+		justifyLeft,
+		justifyCenter,
+		justifyRight,
+		justifySpaceBetween,
+	};
+
 	const [ isPlaceholderShown, setIsPlaceholderShown ] = useState(
 		! hasExistingNavItems
 	);
 
 	const { selectBlock } = useDispatch( 'core/block-editor' );
 
+	// We replace B with -b to allow spaceBetween to become space-between as a classname.
 	const blockProps = useBlockProps( {
 		className: classnames( className, {
-			[ `items-justified-${ attributes.itemsJustification }` ]: attributes.itemsJustification,
+			[ `items-justified-${ replace(
+				attributes.itemsJustification,
+				'B',
+				'-b'
```

Comment:
This is fine, but, as a nitpick, I would do it when getting the icon name instead of here. I think it's easier to read there. So, leave this as: 

```[ `items-justified-${ attributes.itemsJustification }` ]: attributes.itemsJustification,```

and update the call to be the normal css prop later in the file.